tests/ecmp/test_fgnhg.py

```diff
@@ -141,168 +139,514 @@ def create_fg_ptf_config(ptfhost, ip_to_port, port_list, bank_0_port, bank_1_por
-
-
-    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.withdraw_nh.log".format(test_time)
-
-    time.sleep(1)
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.second_link_up.log".format(test_time)
 
     ptf_runner(ptfhost,
             "ptftests",
             "fg_ecmp_test.FgEcmpTest",
             platform_dir="ptftests",
-            params={"test_case": 'withdraw_nh',
+            params={"test_case": 'add_nh',
+                "dst_ip": dst_ip,
                 "config_file": FG_ECMP_CFG,
                 "exp_flow_count": exp_flow_count,
-                "withdraw_nh_port": withdraw_nh_port},
+                "add_nh_port": shutdown_link},
             qlen=1000,
             log_file=log_file)
 
 
-    exp_flow_count = {}
-    for port in port_list:
-        exp_flow_count[port] = flows_per_nh
+    ### Send the same flows again, but enable the next-hop which was down previously
+    ### and check flow hash redistribution
+    logger.info("Send the same flows again, but enable the next-hop which was down previously "
+                " and check flow hash redistribution")
 
+    cmd = vtysh_base_cmd
     for nexthop, port in ip_to_port.items():
         if port == withdraw_nh_port:
-            duthost.shell("vtysh -c 'configure terminal' -c '{} {} {}'".format(ipcmd, prefix, nexthop))
-
+            cmd = cmd + " -c '{} {} {}'".format(ipcmd, prefix, nexthop)
+    configure_dut(duthost, cmd)
+    time.sleep(3)
 
-    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.add_nh.{}.log".format(test_time)
+    exp_flow_count = {}
+    flows_per_nh = NUM_FLOWS/len(port_list)
+    for port in port_list:
+        exp_flow_count[port] = flows_per_nh
 
-    time.sleep(1)
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.add_nh.log".format(test_time)
 
     ptf_runner(ptfhost,
             "ptftests",
             "fg_ecmp_test.FgEcmpTest",
             platform_dir="ptftests",
             params={"test_case": 'add_nh',
+                "dst_ip": dst_ip,
                 "config_file": FG_ECMP_CFG,
                 "exp_flow_count": exp_flow_count,
                 "add_nh_port": withdraw_nh_port},
             qlen=1000,
             log_file=log_file)
 
 
-    withdraw_nh_bank = bank_0_port
-    for nexthop, port in ip_to_port.items():
-        if port in withdraw_nh_bank:
-            duthost.shell("vtysh -c 'configure terminal' -c 'no {} {} {}'".format(ipcmd, prefix, nexthop))
+    ### Simulate route and link flap conditions by toggling the route
+    ### and ensure that there is no orch crash and data plane impact
+    logger.info("Simulate route and link flap conditions by toggling the route "
+                "and ensure that there is no orch crash and data plane impact")
+    nexthop_to_toggle = ip_to_port.keys()[0]
 
+    cmd = "for i in {1..50}; do "
+    cmd = cmd + vtysh_base_cmd
+    cmd = cmd + "  -c 'no {} {} {}';".format(ipcmd, prefix, nexthop_to_toggle)
```

Comment:
we can use API `get_vtysh_cmd_for_namespace()`

Reply:
Can this improvement be taken as part of next iteration of the test? I would prefer having this PR checked in due to the length of time it has been open. The next PR will be opened shortly after this by Kavin.

Reply:
Added the same, thanks, vtysh_base_cmd is now formed via get_vtysh_cmd_for_namespace